lib/puppetfile-resolver/util.rb

```diff
@@ -19,5 +19,27 @@ def self.symbolise_object(object)
     def self.static_ca_cert_file
       @static_ca_cert_file ||= File.expand_path(File.join(__dir__, 'data', 'ruby_ca_certs.pem'))
     end
+
+    # Executes the block with the given proxy set in ENV.
+    def self.with_proxy(new_proxy)
```

Comment:
This is pulled directly from r10k to keep the behavior for setting the proxy the same. I wasn't sure if this was better suited in `PuppetfileResolver::Util`, `PuppetfileResolver::SpecSearchers::Common`, or `PuppetfileResolver::ResolutionProvider`, so dropped it here for now.